app_test.go

```diff
@@ -28,43 +28,42 @@ import (
 	"testing"
 	"time"
 
+	. "go.uber.org/fx"
+	"go.uber.org/fx/fxtest"
+
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-func TestNewApp(t *testing.T) {
-	t.Run("InitializesFields", func(t *testing.T) {
-		app := New()
-		assert.NotNil(t, app.container)
-		assert.NotNil(t, app.lifecycle)
-		assert.NotNil(t, app.logger)
-	})
+type printerSpy struct {
```

Comment:
Possibly, yeah! For now, though, I'd prefer not to encourage external users to test Fx log output, since we'll break a bunch of tests if we change anything.